includes/emails/class-wcs-email-customer-notification.php

```diff
@@ -65,6 +74,31 @@ public function init_form_fields() {
 		);
 	}
 
+	/**
+	 * Update message parameter for the redirect URL when redirecting after
+	 * an unsuccessful order (subscription) action to send a notification.
+	 *
+	 * @param string $redirect_to
+	 * @param int $subscription_id
+	 * @param WC_Subscription $subscription
+	 *
+	 * @return mixed|string
+	 */
+	public function filter_order_redirect( $redirect_to, $subscription_id, $subscription ) {
+
+		if ( ! wcs_is_subscription( $subscription ) ) {
+			return $redirect_to;
+		}
+
+		// Replace 'message=1' with 'message=11' to display correct info to the user.
+		if ( strpos( $redirect_to, 'message=1' ) !== false ) {
+			$redirect_to = remove_query_arg( 'message', $redirect_to );
+			$redirect_to = add_query_arg( 'message', '11', $redirect_to );
+		}
+
+		return $redirect_to;
+	}
```

Comment:
This approach sadly won't work for non-HPOS stores.

Thinking more on this, did we want to add an order note on the subscription for these manual attempts that failed? Similar to the one that we add when the scheduled attempt fails?

Adding some sort of order note for manually triggered actions is common within Subscriptions so it might just make more sense to add a notice and not worry about this "Subscription updated." notice that is shown.

---

If we want to keep exploring other ideas, instead of using this approach with modifying the message param and creating a new messsage type when registering the subscription type, another approach could be to add a notice ourselves and handle the redirect yourself. For example, inside the `trigger()` you could do something like:

```
if ( ! $this->is_enabled()
	|| ! $this->get_recipient()
	|| ! WC_Subscriptions_Email_Notifications::should_send_notification()
	|| WCS_Action_Scheduler_Customer_Notifications::is_subscription_period_too_short( $subscription )
) {
	if ( WC_Subscriptions_Admin::is_edit_subscription_page() ) {
		wcs_add_admin_notice( sprintf( __( 'Attempt to send "%1$s" to %2$s failed.', 'woocommerce-subscriptions' ), $this->title, $this->recipient ), 'error' );
		wp_safe_redirect( $subscription->get_edit_order_url() );
		exit;
	}
	return;
}
```

> [!NOTE]
> `WC_Subscriptions_Admin::is_edit_subscription_page()` is currently a private static function, but we can make this public.

Reply:
Thanks for catching that, I wasn't aware. Converted it to an order note
<img width="319" alt="Screenshot 2024-11-05 at 12 10 36" src="https://github.com/user-attachments/assets/da24f111-2edf-404e-9340-9f96b93179ed">